File: Leetcode/Python3/Array/removeDuplicates.py
# -*- coding:UTF-8 -*-
'''
Solution:
- 双指针
'''
class Solution:
    def removeDuplicates(self, nums: 'List[int]') -> 'int':
        if len(nums) < 2:
            return len(nums)
        i = 0
        for j in range(1, len(nums)):
            if nums[j] == nums[i]:
                continue
            else:
                i += 1
                nums[i] = nums[j]
        return i + 1

File: Leetcode/C++/TwoPointer/removeDuplicates.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size() < 2){
            return nums.size();
        }
        int i = 0, j = i + 1;
        while(j < nums.size()){
            if(nums[i] == nums[j]){
                j++;
            }
            else{
                nums[++i] = nums[j];
            }
        }
        return i+1;
    }
};